aptos-move/aptos-vm/src/aptos_vm.rs

```diff
@@ -217,6 +219,9 @@ impl AptosVM {
         let _timer = TIMER.timer_with(&["AptosVM::new"]);
         let randomness_api_v0_required_deposit = RequiredGasDeposit::fetch_config(resolver)
             .unwrap_or_else(RequiredGasDeposit::default_if_missing);
+        let allow_rand_txn_custom_max_gas = AllowCustomMaxGasFlag::fetch_config(resolver)
```

Comment:
Im just leaning to putting related configs close to each other.

Reply:
I think features are cleaner. Imagine we have more and more features and everyone adds a new struct for those... this can quickly become a mess. Either you have a config stored as Vec<u8>, e.g. gas schedule, which has multiple fields which evolve, or a feature flag. Otherwise this stays forever...

cc @vgao1996 @msmouse who are feature-pros

Reply:
I prefer a normal feature flag in this case as well.

Reply:
if the flag was controlling an entire standalone feature, it makes more sense for the flag to live in the feature flag vec.

But in this case, technically it is part of another config which has been separated out.

Reply:
I also found the features flag vec is very limited: it only support boolean flags, but more and more features need configs like this:
```
enum FeatureXConfig {
    Mode0,
    Mode1 { limit_a: u64 },
    Mode2 { limit_a: u64, threshold_b: u64 },
}
```

Reply:
Feature flags are only for booleans, yes. If you need the config like you want above, with options and fields, there are two ways:

1) Define a rust struct which is initialised based on boolean flags. Please check how `VMConfig` or `VerifierConfig` are created. They have built-in values which we can upgrade per release, and also use feature flags to make stricter.
2) Have a config stored on chain as a serialized vector blob. Our gas schedule is exactly that, it is just an enum of BTreeMaps. This way new versions can be added, and new gas params. Another example: https://github.com/aptos-labs/aptos-core/blob/c729bc3f73de2aa9ba0b228679796d70d6e233b0/aptos-move/aptos-vm-types/src/storage/io_pricing.rs#L234 initialized based on gas feature version. Another example: https://github.com/aptos-labs/aptos-core/blob/c729bc3f73de2aa9ba0b228679796d70d6e233b0/types/src/block_executor/config.rs#L21, but seems like it is not initialized from on-chain yet. But this is a good one: https://github.com/aptos-labs/aptos-core/blob/c729bc3f73de2aa9ba0b228679796d70d6e233b0/types/src/on_chain_config/execution_config.rs#L12.